### PR TITLE
[Android] use view animation for location spinner instead of frame-by-frame animation

### PR DESCRIPTION
**Pull request by RedAuburn**
Gets rid of loads of redundant drawables, and looks smoother on high refresh rate displays

looks identical (other than maybe being slightly smoother)

Saves 762Kb (measured)

### CODE REVIEW

File: android/app/src/main/java/app/organicmaps/widget/menu/MyPositionButton.java [hunk 1]
```diff
@@ -77,8 +80,20 @@ public void update(int mode)
     ImageViewCompat.setImageTintList(mButton, ColorStateList.valueOf(ThemeUtils.getColor(context, colorAttr)));
     updatePadding(mode);
 
-    if (image instanceof AnimationDrawable)
-      ((AnimationDrawable) image).start();
+    if (mode == LocationState.PENDING_POSITION)
+    {
+      final RotateAnimation rotate = new RotateAnimation(0, 360,
+              Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF,
+              0.5f);
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/widget/menu/MyPositionButton.java:87]: The formatting was strange... A bit longer lines (120-140 chars) are perfectly ok, they're easier to read on modern screens.
```suggestion
      final RotateAnimation rotate = new RotateAnimation(0, 360, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f);
```

**Review by biodranik — APPROVED** [review]
Thanks! How does it look visually? Does it work on Android 5? CC @Jean-BaptisteC

**Comment by RedAuburn** [comment]
(I did try to convert the icon to SVG, but apparently it's actually impossible to draw a conical gradient in SVG, like, not part of the spec)

File: android/app/src/main/java/app/organicmaps/widget/menu/MyPositionButton.java (hunk 1 above)
Comment (RedAuburn) [reply android/app/src/main/java/app/organicmaps/widget/menu/MyPositionButton.java]: fixed, turns out `Animation.RELATIVE_TO_SELF` is just = `1`, so it all fits on one line anyway 😅

**Review by Jean-BaptisteC — APPROVED** [review]
✅ Pixel 6 - Android 14 and Honor 8 - Android 8
My tablet on Android 5 is dead :(

File: android/app/src/main/java/app/organicmaps/widget/menu/MyPositionButton.java (hunk 1 above)
Comment (biodranik) [reply android/app/src/main/java/app/organicmaps/widget/menu/MyPositionButton.java]: Using descriptive text constants is more readable than just numbers )

Comment (RedAuburn) [reply android/app/src/main/java/app/organicmaps/widget/menu/MyPositionButton.java]: fair enough, fixed :)

**Comment by biodranik** [comment]
Was it tested in Android 5 Emulator? @RedAuburn @Jean-BaptisteC

**Comment by RedAuburn** [comment]
✅ Oneplus One - Android 5.1.1

**Pull request merged**